source/opt/type_manager.cpp

```diff
@@ -636,15 +648,39 @@ Type* TypeManager::RecordIfTypeDefinition(const Instruction& inst) {
     case SpvOpTypeSampledImage:
       type = new SampledImage(GetType(inst.GetSingleWordInOperand(0)));
       break;
-    case SpvOpTypeArray:
-      type = new Array(GetType(inst.GetSingleWordInOperand(0)),
-                       inst.GetSingleWordInOperand(1));
+    case SpvOpTypeArray: {
+      const uint32_t length_id = inst.GetSingleWordInOperand(1);
+      const Instruction* length_constant_insn = id_to_constant_insn_[length_id];
```

Comment:
I would prefer to use "inst" as the short form for "instruction".  That is already commonly used in spirv-tools.